Cython/Compiler/Dataclass.py

```diff
@@ -271,7 +270,7 @@ def handle_cclass_dataclass(node, dataclass_args, analyse_decs_transform):
     # generic objects and thus can accept _HAS_DEFAULT_FACTORY.
     # Type conversion comes later
     comp_directives = Nodes.CompilerDirectivesNode(node.pos,
-        directives=copy_inherited_directives(node.scope.directives, annotation_typing=False),
+        directives=dict(**node.scope.directives, annotation_typing=False),
```

Comment:
This probably just fails because it receives the keyword argument twice. Remove the stars and pass the plain dict for copying, then it should work.
`directives=dict(node.scope.directives, annotation_typing=False)`

Reply:
That definitely was an issue (and I tried something similar to what you suggested in https://github.com/cython/cython/pull/4575/commits/84822a6db69f14459ea12601eafd3454952c9ad8).

Ultimately the real issue was that the `test_fail_if_path_exists` etc ended up being copied into these compiler directives nodes, and then causing test failures. So I came to the conclusion that `copy_inherited_directives` was serving a useful purpose.

I'm still not completely sure they should be needed, but it didn't seem like an easy fix to remove them

Reply:
Either way, I think it's better to have a dedicated function for this, rather than spreading the assumption throughout the code base that calling `dict()` is enough.